game_eggs/tycoon_games/openrct2/README.md

```diff
@@ -0,0 +1,88 @@
+# OpenRCT2
+***Note:** OpenRCT2 only recently (Jan. 2022) updated to support non-interactive Docker environments, and development is ongoing to improve it. Please understand future updates may be unstable, or this Egg may update in the future to accommodate changes.*
+___
+### Authors / Contributors
+<!-- prettier-ignore-start -->
+<!-- markdownlint-disable -->
+<table>
+    <tr>
+        <td align="center">
+            <a href="https://github.com/lilkingjr1">
+                <img src="https://avatars.githubusercontent.com/u/4533989" width="50px;" alt=""/><br /><sub><b>Red-Thirten</b></sub>
+            </a>
+            <br />
+            <a href="https://github.com/parkervcp/eggs/commits?author=lilkingjr1" title="Codes">💻</a>
+            <a href="https://github.com/parkervcp/eggs/commits?author=lilkingjr1" title="Maintains">🔨</a>
+        </td>
+        <td align="center">
+            <a href="https://github.com/parkervcp">
+                <img src="https://avatars.githubusercontent.com/u/1207679" width="50px;" alt=""/><br /><sub><b>parkervcp</b></sub>
+            </a>
+            <br />
+            <a href="https://github.com/parkervcp/eggs/commits?author=parkervcp" title="Codes">💻</a>
+            <a href="https://github.com/parkervcp/eggs/commits?author=parkervcp" title="Contributor">💡</a>
+        </td>
+        <td align="center">
+            <a href="https://github.com/janisozaur">
+                <img src="https://avatars.githubusercontent.com/u/550290" width="50px;" alt=""/><br /><sub><b>janisozaur</b></sub>
+            </a>
+            <br />
+            <a href="https://github.com/OpenRCT2/OpenRCT2/commits?author=janisozaur" title="Codes">💻</a>
+            <a href="https://github.com/OpenRCT2/OpenRCT2/commits?author=janisozaur" title="Contributor">💡</a>
+        </td>
+    </tr>
+</table>
+<!-- markdownlint-enable -->
+<!-- prettier-ignore-end -->
+
+___
+### Game Description
+From OpenRCT2's [Website](https://openrct2.org/):
+> OpenRCT2 is an open-source re-implementation of RollerCoaster Tycoon 2 (RCT2), expanding the game with new features, fixing bugs and raising game limits. The gameplay revolves around building and maintaining an amusement park containing attractions, shops and facilities. The player must try to make a profit and maintain a good park reputation whilst keeping the guests happy. OpenRCT2 allows for both scenario and sandbox play. Scenarios require the player to complete a certain objective in a set time limit whilst sandbox allows the player to build a more flexible park with optionally no restrictions or finance.
+___
+### Server Ports
+| Port | Default (TCP) |
+|---------|---------|
+| Game | 11753 |
+
+___
+### Installation/System Requirements
+
+|  | Minimum |
+|---------|---------|
+| Processor | Recent x86/64 (AMD/Intel) processor. No ARM support. |
```

Comment:
Can we mention here upstream (the OpenRCT2 project) supports ARM builds? Given this egg is compiled from source, I don't even see problems with enabling such support here - can I get your insight what needs to be done?

Reply:
The docker image does not support ARM64. It's a generic Debian image with packages for Steam, which don't exist on ARM64

Reply:
Could write a custom image for OpenRCT2 that supports ARM, but I'm under the impression that the ethos for Pterodactyl eggs is to use a generic image if at all possible. So unless a custom image is approved, it will likely stay as is.

Reply:
Where are the images' definitions? I see no x86-specific packages here: https://github.com/pterodactyl/yolks/blob/master/oses/debian/Dockerfile or https://github.com/parkervcp/yolks/blob/master/oses/debian/Dockerfile, but I'm probably looking at wrong files.

The "one size fits all" approach seems to be at odds with docker's goal of providing minimal images that only ship with necessary components to run particular service. OpenRCT2 needs no Steam and should not be run in an image that contains it.

Reply:
https://github.com/parkervcp/yolks/blob/master/.github/workflows/base.yml

We're generally against game-specific images unless existing images can't run it, as it would require maintaining a lot of images, instead of one that a lot of servers can share without it pulling and wasting space. One 300MB image vs 50 of 100MB each.

ARM64 support itself with Pterodactyl is new.

Reply:
So it seems I'm looking at correct `Dockerfile`? I'm confused by your earlier statement about "generic Debian image with packages for Steam", as I can't find them.

https://github.com/parkervcp/yolks/blob/8a4ef8387de35150c1b8f0ce9bf6c308d2e9f595/.github/workflows/base.yml#L36 uses https://github.com/parkervcp/yolks/blob/8a4ef8387de35150c1b8f0ce9bf6c308d2e9f595/.github/workflows/base.yml#L20 and points to https://github.com/parkervcp/yolks/blob/master/oses/debian/Dockerfile and there are no steam-specific packages.

> pulling and wasting space

```
FROM ghcr.io/parkervcp/yolks:debian-base
RUN apt install steam-stuff
```

Reply:
The steam-specific package was `lib32gcc-s1`. I've removed it and made all the base images be multi-arch, thus openrct can now also run on ARM64.

Reply:
:tada: